core/comment/service.go

```diff
@@ -0,0 +1,168 @@
+package comment
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/goto/guardian/domain"
+	"github.com/goto/guardian/pkg/log"
+	"github.com/goto/guardian/plugins/notifiers"
+)
+
+const (
+	AuditKeyCreate = "comment.create"
+)
+
+var (
+	ErrEmptyCommentCreator = errors.New("comment creator (\"created_by\") can't be empty")
+	ErrEmptyCommentBody    = errors.New("comment can't be empty")
+)
+
+//go:generate mockery --name=repository --exported --with-expecter
+type repository interface {
+	Create(context.Context, *domain.Comment) error
+	List(context.Context, domain.ListCommentsFilter) ([]*domain.Comment, error)
+}
+
+//go:generate mockery --name=appealService --exported --with-expecter
+type appealService interface {
```

Comment:
agree, @idilhaq let's remove this interface and use the appealservice struct type directly. We might need to update test cases as well to initialize an actual `appealService` along with its dependencies (some can be mocked e.g. repository) instead of using `mockAppealService`